src/gt4py/next/iterator/transforms/global_tmps.py

```diff
@@ -113,9 +119,53 @@ def pformat_FencilWithTemporaries(
 
 
 # Main implementation
+def canonicalize_applied_lift(closure_params: list[str], node: ir.Node) -> ir.Node:
+    """
+    Canonicalize applied lift expressions.
 
-
-def split_closures(node: ir.FencilDefinition) -> FencilWithTemporaries:
+    >>> expr = im.lift(im.lambda_("a")(im.deref("a")))(im.lift("deref")("inp"))
+    >>> print(expr)
+    (↑(λ(a) → ·a))((↑deref)(inp))
+    >>> print(canonicalize_applied_lift(["inp"], expr))
+    (↑(λ(inp) → (λ(a) → ·a)((↑deref)(inp))))(inp)
+    """
+    assert (
+        isinstance(node, ir.FunCall)
+        and isinstance(node.fun, ir.FunCall)
+        and node.fun.fun == ir.SymRef(id="lift")
+    )
+    stencil = node.fun.args[0]
+    it_args = node.args
+    if any(not isinstance(it_arg, ir.SymRef) for it_arg in it_args):
+        used_closure_params = collect_symbol_refs(node)
+        assert not (set(used_closure_params) - set(closure_params))
+        return im.lift(im.lambda_(*used_closure_params)(im.call(stencil)(*it_args)))(
+            *used_closure_params
+        )
+    return node
+
+
+def temporary_extraction_predicate(expr: ir.Node, num_occurences: int) -> bool:
+    """Determine if `expr` is an applied lift that should be extracted as a temporary."""
+    if not (
+        isinstance(expr, ir.FunCall)
+        and isinstance(expr.fun, ir.FunCall)
+        and expr.fun.fun == ir.SymRef(id="lift")
+    ):
+        return False
```

Comment:
Let's do the refactoring of extracting `is_applied_lift` to a shared location from all passes. (if you like to do it in a follow up pr, fine, but then add a TODO